ext/mysqli/mysqli.stub.php

```diff
@@ -0,0 +1,1303 @@
+ */
+function mysqli_real_connect(mysqli $mysqli_link, ?string $host = null, ?string $user = null, ?string $password = null, ?string $database = null, ?int $port = null, ?int $flags = null): mysqli {}
+
+/**
+ * @param mysqli $mysqli_link
+ * @param string $escapestr
+ *
+ * @return string
+ */
+function mysqli_real_escape_string(mysqli $mysqli_link, string $escapestr): string {}
+
+/**
+ * @param mysqli $mysqli_link
+ * @param string $query
+ *
+ * @return bool
+ */
+function mysqli_real_query(mysqli $mysqli_link, string $query): bool {}
+
+/**
+ * @param mysqli $mysqli_link
+ *
+ * @return mysqli_result|bool
+ */
+function mysqli_reap_async_query(mysqli $mysqli_link) {}
+
+/**
+ * @param mysqli $mysqli_link
+ * @param string $name
+ *
+ * @return bool
+ */
+function mysqli_release_savepoint(mysqli $mysqli_link, string $name): bool {}
+
+/**
+ * @param mysqli $mysqli_link
+ * @param int $flags
+ * @param string $name
+ *
+ * @return bool
+ */
+function mysqli_rollback(mysqli $mysqli_link, int $flags = 0, string $name = ''): bool {}
+
+/**
+ * @param mysqli $mysqli_link
+ * @param string $name
+ *
+ * @return bool
+ */
+function mysqli_savepoint(mysqli $mysqli_link, string $name): bool {}
+
+/**
+ * @param mysqli $mysqli_link
+ * @param string $dbname
+ *
+ * @return bool
+ */
+function mysqli_select_db(mysqli $mysqli_link, string $dbname): bool {}
+
+/**
+ * @param mysqli $mysqli_link
+ * @param string $cs_name
+ *
+ * @return bool
+ */
+function mysqli_set_charset(mysqli $mysqli_link, string $cs_name): bool {}
+
+/**
+ * @param mysqli_stmt $mysql_stmt
+ *
+ * @return int
+ */
+function mysqli_stmt_affected_rows(mysqli_stmt $mysql_stmt): int {}
+
+/**
+ * @param mysqli_stmt $mysql_stmt
+ * @param int $attr
+ *
+ * @return int|bool
+ */
+function mysqli_stmt_attr_get(mysqli_stmt $mysql_stmt, int $attr) {}
+
+/**
+ * @param mysqli_stmt $mysql_stmt
+ * @param int $attr
+ * @param int $mode_in
+ *
+ * @return bool
+ */
+function mysqli_stmt_attr_set(mysqli_stmt $mysql_stmt, int $attr, int $mode_in): bool {}
+
+/**
+ * @param mysqli_stmt $mysql_stmt
+ * @param string $types
+ * @param mixed &...$vars
+ *
+ * @return bool
+ */
+function mysqli_stmt_bind_param(mysqli_stmt $mysql_stmt, string $types, mixed &...$vars): bool {}
+
+/**
+ * @param mysqli_stmt $mysql_stmt
+ * @param mixed &...$vars
+ *
+ * @return bool
+ */
+function mysqli_stmt_bind_result(mysqli_stmt $mysql_stmt, mixed &...$vars): bool {}
+
+/**
+ * @param mysqli_stmt $mysql_stmt
+ *
+ * @return bool
+ */
+function mysqli_stmt_close(mysqli_stmt $mysql_stmt): bool {}
+
+/**
+ * @param mysqli_stmt $mysql_stmt
+ * @param int $offset
+ *
+ * @return void|bool
+ */
+function mysqli_stmt_data_seek(mysqli_stmt $mysql_stmt, int $offset) {}
+
+/**
+ * @param mysqli_stmt $mysql_stmt
+ *
+ * @return int
+ */
+function mysqli_stmt_errno(mysqli_stmt $mysql_stmt): int {}
+
+/**
+ * @param mysqli_stmt $mysql_stmt
+ *
+ * @return string
+ */
+function mysqli_stmt_error(mysqli_stmt $mysql_stmt): string {}
```

Comment:
The phpDocs for methods/functions that already have native types declare is not necessary. Feel free to drop them 👍

Reply:
- rebase ok
- phpDocs is removed where the signature was identical

I hope I didn't break anything with this PR. But the tests seemed to pass locally :)

Reply:
@tvlooy, CI will show (tests can only fail in debug mode, though)

Reply:
many fails :( Rebuilding with --enable-debug and doing rework

Reply:
Off the top of my head: there are a few macros which are basically used in all functions, and which may return FALSE or NULL (so there likely have to be many ?bool returns, etc.)

Reply:
Should I just leave the return type empty if too many types are in there or should I try to add all types? (the latter is what I did but seems less desirable, but test pass)

Reply:
I think it's best to explicit and precise here.